controllers/workspace/provision/deployment.go

```diff
@@ -107,6 +109,35 @@ func SyncDeploymentToCluster(
 		}
 	}
 
+	// Get pod and update dw status on crashloopbackoff
+	podList, err := getPods(workspace, clusterAPI.Client)
+	if err != nil {
+		return DeploymentProvisioningStatus{
+			ProvisioningStatus: ProvisioningStatus{
+				Requeue: true,
+				Err:     err,
+			},
+		}
+	}
+
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodPending {
+			return DeploymentProvisioningStatus{
+				ProvisioningStatus: ProvisioningStatus{Requeue: true},
+			}
+		}
+
+		for _, containerStatuses := range pod.Status.ContainerStatuses {
+			if containerStatuses.State.Waiting != nil && containerStatuses.State.Waiting.Reason == constants.ContainerCrashLoopBackOffReason {
```

Comment:
This is an issue for a separate PR, but it'd be great to eventually check reasons against a list of "failure" states, e.g. `CrashLoopBackOff`, `ImagePullErr`, etc.

Reply:
I faced this issue today. Due `ImagePullErr` my workspace was starting forever.
We should fail workspace in such case or at least propagate it as a message.

Reply:
done!